### PR TITLE
Update Install for Pytorch version

### PR DESCRIPTION
**Pull request by Fhrozen**
REF: #1631 

Installation checking for new pytorch versions.

BTW, I also quiet pytest. I am not sure if it is fine. IMO, the only important details are the failed tests and this option does not affect them. Showing which test has passes generates a really long list and now even larger because of the multiple tests. Let me know if this is fine or if I should reverse it.

### CODE REVIEW

**Comment by Fhrozen** [comment]
@sw005320 ready. Maybe need @kamo-naoyuki @kan-bayashi @ShigekiKarita overview?

File: espnet/nets/pytorch_backend/ctc.py [hunk 1]
```diff
@@ -22,7 +22,10 @@ def __init__(self, odim, eprojs, dropout_rate, ctc_type='warpctc', reduce=True):
         self.dropout_rate = dropout_rate
         self.loss = None
         self.ctc_lo = torch.nn.Linear(eprojs, odim)
-        self.ctc_type = ctc_type
+
+        # In case of Pytorch >= 1.2.0, CTC will be always builtin
+        torch_ver = int(torch.__version__.replace('.', '').replace('post2', ''))
```
Comment (kan-bayashi) [inline espnet/nets/pytorch_backend/ctc.py]: It is better to use LooseVersion (see setup.py for example).
Also it is better to add logging.warning for notification if ctc_type is different.

File: tools/Makefile [hunk 2]
```diff
@@ -6,8 +6,13 @@ PYTHON :=
 PYTHON_VERSION := 3.7.3
 CUPY_VERSION := 6.0.0
 CUDA_VERSION := 10.0
-# PyTorch version: 0.4.1 or 1.0.0 or 1.0.1
+# PyTorch version: 0.4.1, 1.0.0, 1.0.1, 1.1.0, 1.2.0, 1.3.0, 1.3.1
```
Comment (kan-bayashi) [inline tools/Makefile]: 1.4 is not working?

File: tools/check_install.py [hunk 3]
```diff
@@ -13,22 +13,34 @@ def main(args):
...
         ('chainer', ("6.0.0")),
         ('chainer_ctc', None),
-        ('warpctc_pytorch', ("0.1.1", "0.1.3")),
         ('warprnnt_pytorch', ("0.1"))
     ]
 
     if not args.no_cupy:
         MANUALLY_INSTALLED_LIBRARIES.append(('cupy', ("6.0.0")))
+    
+    torch_ver = int(args.torch_version.replace('.', '').replace('post2', ''))
```
Comment (kan-bayashi) [inline tools/check_install.py]: ditto

File: tools/Makefile (hunk 2 above)
Comment (Fhrozen) [reply tools/Makefile]: I did not test the version but should be working if there is no big difference from 1.3.1

Comment (sw005320) [reply tools/Makefile]: @Emrys365 already tested and it's working (see https://github.com/espnet/espnet/pull/1527)

**Comment by sw005320** [comment]
Thanks!

**Comment by qmpzzpmq** [comment]
hi, According your previous benchmark of different ctc, it seems that the result of model training from warpctc is better than from builtinctc.
Have you guys test these benchmark in these newer versions pytorch. It seems that I found the built in ctc is much faster than waprtctc especially in with accerlerating.

**Comment by sw005320** [comment]
We tested it with several pytorch versions, and our conclusion is that warpctc is slightly better but we don't observe so much difference in various builtin pytorch CTC versions. We did not check the speed though. Could you share your speed results with us?

**Comment by qmpzzpmq** [comment]
@sw005320 I am testing recently, but I fund there is some conflict between apex with builtin ctc.

**Comment by sw005320** [comment]
Could you report that conflict? I'm curious.

**Comment by sw005320** [comment]
Sorry I saw your report in #1751 
Many thanks!
I'll take a look at it.

**Pull request merged**